src/main/java/com/splunk/hecclient/Indexer.java

```diff
@@ -137,14 +156,60 @@ public boolean send(final EventBatch batch) {
     @Override
     public synchronized String executeHttpRequest(final HttpUriRequest req) {
         CloseableHttpResponse resp;
-        try {
-            resp = httpClient.execute(req, context);
-        } catch (Exception ex) {
-            logBackPressure();
-            log.error("encountered io exception", ex);
-            throw new HecException("encountered exception when post data", ex);
+        if (hecConfig.kerberosAuthEnabled()) {
+            if (config == null) {
```

Comment:
extract this to a separate method defineKerberosConfigs(...)
cause this function is supposed to just execute the http request.